internal/cmd/beta/kms/version/disable/disable.go

```diff
@@ -0,0 +1,149 @@
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(params.Printer, params.CliVersion)
+			if err != nil {
+				return err
+			}
+
+			// This operation can be undone. Don't ask for confirmation!
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			err = req.Execute()
+			if err != nil {
+				return fmt.Errorf("disable key version: %w", err)
+			}
+
+			// kms v1.0.0 has a waiter, but it get's stuck even though the disable api call was already successfully completed.
+
+			// Get the key version in its state afterwards
+			resp, err := apiClient.GetVersionExecute(ctx, model.ProjectId, model.Region, model.KeyRingId, model.KeyId, model.VersionNumber)
+			if err != nil {
+				params.Printer.Debug(print.ErrorLevel, "get key version: %v", err)
+			}
+
+			return outputResult(params.Printer, model.OutputFormat, resp)
+		},
+	}
+
+	configureFlags(cmd)
+	return cmd
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command, inputArgs []string) (*inputModel, error) {
+	versionStr := inputArgs[0]
+	versionNumber, err := strconv.ParseInt(versionStr, 10, 64)
+	if err != nil || versionNumber < 0 {
+		return nil, &errors.ArgValidationError{
+			Arg:     versionNumberArg,
+			Details: fmt.Sprintf("invalid value %q: must be a positive integer", versionStr),
+		}
+	}
+
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		KeyRingId:       flags.FlagToStringValue(p, cmd, keyRingIdFlag),
+		KeyId:           flags.FlagToStringValue(p, cmd, keyIdFlag),
+		VersionNumber:   versionNumber,
+	}
+
+	p.DebugInputModel(model)
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *kms.APIClient) kms.ApiDisableVersionRequest {
+	return apiClient.DisableVersion(ctx, model.ProjectId, model.Region, model.KeyRingId, model.KeyId, model.VersionNumber)
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), keyRingIdFlag, "ID of the KMS key ring")
+	cmd.Flags().Var(flags.UUIDFlag(), keyIdFlag, "ID of the key")
+
+	err := flags.MarkFlagsRequired(cmd, keyRingIdFlag, keyIdFlag)
+	cobra.CheckErr(err)
+}
+
+func outputResult(p *print.Printer, outputFormat string, resp *kms.Version) error {
+	if resp == nil {
+		return fmt.Errorf("response is nil")
+	}
+
+	switch outputFormat {
+	case print.JSONOutputFormat:
+		details, err := json.MarshalIndent(resp, "", "  ")
+		if err != nil {
+			return fmt.Errorf("marshal KMS key: %w", err)
+		}
+		p.Outputln(string(details))
+
+	case print.YAMLOutputFormat:
+		details, err := yaml.MarshalWithOptions(resp, yaml.IndentSequence(true), yaml.UseJSONMarshaler())
+		if err != nil {
+			return fmt.Errorf("marshal KMS key: %w", err)
+		}
+		p.Outputln(string(details))
+
+	default:
+		p.Outputf("Disabled version %d of the key '%s'\n", utils.PtrValue(resp.Number), utils.PtrValue(resp.KeyId))
```

Comment:
nitpick: `'%s'` can be replaced by `%q` to set a quoted string :)
```suggestion
		p.Outputf("Disabled version %d of the key %q\n", utils.PtrValue(resp.Number), utils.PtrValue(resp.KeyId))
```